### DISCUSSION TITLE
Incremental source generator VB support

### DISCUSSION THREAD

**Discussion by weifenluo** [dotnet/roslyn#67860] [Q&A] [answered]
Does VB support incremental source generator? I'm asking this question because I just found out that `VisualBasicGeneratorDriver.Create` only takes `ISourceGenerator` as input param.

**Comment by jasonmalinowski** [comment, marked as answer]
Oops, I guess we have the helper method on the C# side but maybe not on the VB side. (Tagging @chsienki to that effect.) What you can do though is for any IIncrementalGenerator, you can call [the AsSourceGenerator extension method](https://github.com/dotnet/roslyn/blob/929789958cb0e0a943fe0d835368136a4e49e4fe/src/Compilers/Core/Portable/SourceGeneration/GeneratorExtensions.cs#L37) which wraps any IIncrementalGenerator in an object that looks like an ISourceGenerator to pass around to any other method. Don't worry, under the covers we still treat it as an incremental generator -- we introduced that API rather than having to update all the generator APIs to be able to deal with either interface type.

**Reply by chsienki** [reply]
Yes, even on the CSharp side, only some of the overloads directly takes an `IIncrementalGenerator` and other you have to wrap it using `AsSourceGenerator()`. I suspect as we move to deprecate `ISourceGenerator` we'll standardize on using `IIncrementalGenerator` and add the matching overloads as needed in one pass.
